cinema/permissions.py

```diff
@@ -0,0 +1,12 @@
+from rest_framework.permissions import (
+    BasePermission,
+    SAFE_METHODS
+)
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+    def has_permission(self, request, view):
+        return bool(
+            request.method in SAFE_METHODS
+            and request.user.is_authenticated
+        ) or request.user.is_staff
```

Comment:
There is an issue with the logical operator precedence in your permissions check. The `and` operator has higher precedence than the `or` operator. This means that your current code checks if the request method is in SAFE_METHODS and the user is authenticated, or if the user is a staff member. If the user is a staff member, they will have permission regardless of the request method or whether they are authenticated. If you want to allow read-only access for authenticated users and full access for staff members, you should add parentheses to change the order of operations. Here's how you can do it: `return bool((request.method in SAFE_METHODS and request.user.is_authenticated) or request.user.is_staff)`.